test/demo/appRestart.inttest.ts

```diff
@@ -0,0 +1,82 @@
+import 'jest';
+
+import {Application, Window} from 'hadouken-js-adapter';
+import {ElementHandle} from 'puppeteer';
+
+import {NotificationOptions, Notification} from '../../src/client';
+
+import * as notifsRemote from './utils/notificationsRemote';
+import {delay, Duration} from './utils/delay';
+import {getToastCards} from './utils/toastUtils';
+import {createApp} from './utils/spawnRemote';
+import {testManagerIdentity, defaultTestAppUrl} from './utils/constants';
+import {waitForAppToBeRunning} from './utils/common';
+
+const relaunchNotificationOptions: NotificationOptions = {
+    body: 'Test Notification Body',
+    title: 'Test Notification Title',
+    category: 'Test Notification Category',
+    customData: {testContext: 'testContext'},
+    onSelect: {task: 'Selected'},
+    buttons: []
+};
+
+const nonRelaunchNotificationOptions: NotificationOptions = {
+    body: 'Test Notification Body',
+    title: 'Test Notification Title',
+    category: 'Test Notification Category'
+};
+
+describe('When an app that uses notification-service is created', () => {
+    let testApp: Application;
+    let testWindow: Window;
+
+    let createPromise: Promise<Notification>;
+    let note: Notification;
+
+    beforeEach(async () => {
+        testApp = await createApp(testManagerIdentity, {url: defaultTestAppUrl});
+        testWindow = await testApp.getWindow();
+    });
+
+    describe('When the application creates a notification and quits', () => {
+        let toastCards: ElementHandle[] | undefined;
+
+        beforeEach(async () => {
+            ({createPromise, note} = await notifsRemote.createAndAwait(testWindow.identity, relaunchNotificationOptions));
+            await delay(Duration.TOAST_DOM_LOADED);
+            toastCards = await getToastCards(testApp.identity.uuid, note.id);
+            await testApp.quit();
+        });
+
+        afterEach(async () => {
+            await notifsRemote.clearAll(testWindow.identity);
+            await testApp.quit();
+        });
```

Comment:
We want `afterEach` to succeed at cleaning up regardless of whether the test itself succeeded, so I think unfortunately this needs to "manually" start the app up again if it's not already running. Also, we either want this in the second `describe` block too, or for this to move up to the top level

Reply:
done